python/guide/packages/gluon/data.md

```diff
@@ -1,325 +1,120 @@
-Linux and Macos are supported.) Let's first benchmark the time to read the MNIST
-training set:
+We can plot a few (augmented) images from the last batch.
 
-```{.python .input  n=7}
-tic = time.time()
-data = gdata.DataLoader(mnist_transformed, batch_size=64)
-for X, y in data:
-    pass
-'%.1f sec' % (time.time() - tic)
 ```
-
-Now let's use 4 processes:
-
-```{.python .input  n=8}
-tic = time.time()
-data = gdata.DataLoader(mnist_transformed, batch_size=64, num_workers=4)
-for X, y in data:
-    pass
-'%.1f sec' % (time.time() - tic)
+plt.figure(1)
+plt.subplot(131); plot(data_batch[123])
+plt.subplot(132); plot(data_batch[234])
+plt.subplot(133); plot(data_batch[345])
+plt.show()
 ```
 
-## Appendix: From `DataIter` to `DataLoader`
-
-Before Gluon's `DataLoader`, MXNet
-provides `DataIter` in the `io` module to read mini-batches. They are similar to
-each other but `DataLoader` often returns a tuple of `(feature, label)` for a
-mini-batches, while `DataIter` returns a `DataBatch`. The following example
-wraps a `DataIter` into a `DataLoader` so you can reuse the existing codes but
-enjoys the benefits of Gluon.
-
-```{.python .input}
-class DataIterLoader():
-    def __init__(self, data_iter):
-        self.data_iter = data_iter
-    def __iter__(self):
-        self.data_iter.reset()
-        return self
-    def __next__(self):
-        batch = self.data_iter.__next__()
-        assert len(batch.data) == len(batch.label) == 1
-        data = batch.data[0]
-        label = batch.label[0]
-        return data, label
-    def next(self):
-        return self.__next__() # for Python 2
-```
+## Conclusion
 
-Now create an `DataIter` instance, and then get the according `DataLoader`
-wrapper.
+We've now seen all of the core components of the MXNet Gluon data pipeline. You should now understand the difference between [`Dataset`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.data.Dataset.html)s, that return individual samples, and [`DataLoader`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.data.DataLoader.html)s, that return batches of samples. You should also be able to transform your `dataset` for pre-processing or augmentation purposes.
 
-```{.python .input}
-data_iter = io.NDArrayIter(data=features, label=labels, batch_size=4)
-data = DataIterLoader(data_iter)
-for X, y in data:
-    print('X shape:', X.shape, '\ty:', y.asnumpy())
-```
+## Recommended Next Steps
+
+We used presets and a lot of default values in this tutorial, but there are many other options and possibilities. You're likely to want to use your own dataset for training. You can learn more about this in the dedicated `Dataset` tutorial. We also have a tutorial on `transform`s with examples on how to compose transform operations. And last but not least, we have a very useful tutorial on `DataLoader`s. You'll often find that loading data can be the bottleneck to training, and this tutorial contains some useful tricks to speed up the data pipeline.
```

Comment:
We mention 3 extra tutorials here: transform, Dataset and Dataloader. Can we provide links to them here?

Reply:
We don't have these right now, but will add links when they're done.